backend/de.metas.util.web/src/main/java/de/metas/util/web/audit/ApiAuditService.java

```diff
@@ -631,6 +636,26 @@ private void handleErrorResponse(
 		buildHttpResponse(httpServletResponse, apiResponse, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
 	}
 
+	@Nullable
+	private static PInstanceId extractPInstanceId(@NonNull final LinkedMultiValueMap<String, String> requestHeadersMultiValueMap)
+	{
+		try
+		{
+			return Optional.ofNullable(requestHeadersMultiValueMap.get(HEADER_PINSTANCE_ID))
+					.filter(pInstanceIdList -> !pInstanceIdList.isEmpty())
+					.map(pInstanceIdList -> pInstanceIdList.get(0))
+					.filter(Check::isNotBlank)
+					.map(Integer::parseInt)
+					.map(PInstanceId::ofRepoId)
+					.orElse(null);
+		}
+		catch (final Exception exception)
+		{
+			logger.error("Exception encountered while trying to read 'x-adpinstanceid' header!", exception);
```

Comment:
as written, imho there should not be and error logged in MF, but the problem should be escalated towards the caller

Reply:
(created a follow-up issue for this)